pkg/kubelet/dockertools/docker.go

```diff
@@ -299,7 +299,12 @@ func ParseDockerName(name string) (dockerName *KubeletContainerName, hash uint64
 }
 
 func LogSymlink(containerLogsDir, podFullName, containerName, dockerId string) string {
-	return path.Join(containerLogsDir, fmt.Sprintf("%s_%s-%s.%s", podFullName, containerName, dockerId, LogSuffix))
+	logPath := fmt.Sprintf("%s_%s-%s.%s", podFullName, containerName, dockerId, LogSuffix)
+	// Length of a filename cannot exceed 256 characters in ext4 on Linux.
+	if len(logPath) > 255 {
+		logPath = logPath[:255]
```

Comment:
@vishh Just an idea, does chopping the chars at the end possibly end up with more collisions? (since dockerId is more unique), should we chop from the beginning?